### PR TITLE
Added ManagedPrefixList

### PR DESCRIPTION
**Pull request by monty16597**
## Added
- Support for `PrefixList` and `ManagedPrefixList`.
- API `CreateManagedPrefixList`
- API `DescribeManagedPrefixLists`
- API `GetManagedPrefixListEntries`
- API `DeleteManagedPrefixList`
- API `DescribePrefixLists`
- API `DeleteVpcEndpoint`
- API `ModifyManagedPrefixList`

## Terraform passed test
- `TestAccAwsEc2ManagedPrefixList_basic`
- `TestAccAwsEc2ManagedPrefixList_disappears`
- `TestAccAwsEc2ManagedPrefixList_AddressFamily_IPv6`
- `TestAccAwsEc2ManagedPrefixList_Entry_Cidr`
- `TestAccAwsEc2ManagedPrefixList_Entry_Description`
- `TestAccAwsEc2ManagedPrefixList_Name`
- `TestAccAwsEc2ManagedPrefixList_Tags`

## Tested with
- Terraform for vpc-endpoint
```terraform
resource "aws_vpc" "main" {
  cidr_block = "10.0.0.0/16"
}

resource "aws_vpc_endpoint" "s3" {
  vpc_id       = aws_vpc.main.id
  service_name = "com.amazonaws.us-west-2.s3"
}

resource "aws_vpc_endpoint" "s32" {
  vpc_id       = aws_vpc.main.id
  service_name = "com.amazonaws.us-west-2.s3"

  tags = {
    Environment = "test"
  }
}

resource "aws_subnet" "main" {
  vpc_id     = aws_vpc.main.id
  cidr_block = "10.0.1.0/24"

  tags = {
    Name = "Main"
  }
}


resource "aws_vpc_endpoint" "ec2" {
  vpc_id            = aws_vpc.main.id
  service_name      = "com.amazonaws.us-west-2.ec2"
  vpc_endpoint_type = "Interface"
  subnet_ids = [aws_subnet.main.id]

  security_group_ids = [
    "sg-kj1h2i3112",
  ]

  private_dns_enabled = true
}
```

- Terraform for managed-prefix-list
```terraform
resource "aws_ec2_managed_prefix_list" "example" {
  name           = "changed"
  address_family = "IPv4"
  max_entries    = 1

  entry {
    cidr        = "10.0.0.0/16"
    description = "testing cidr"
  }
  tags = {
    Env = "live"
  }
}
```

### CODE REVIEW

File: moto/ec2/utils.py [hunk 1]
```diff
@@ -230,6 +231,14 @@ def generate_route_id(route_table_id, cidr_block, ipv6_cidr_block=None):
     return "%s~%s" % (route_table_id, cidr_block)
 
 
+def generate_vpc_end_point_id(vpc_id):
```
Comment (bblommers) [inline moto/ec2/utils.py:234]: This doesn't seem to be used anywhere in this PR?

File: terraform.tfstate [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (bblommers) [inline terraform.tfstate:1]: This file shouldn't be part of Moto

**Review by bblommers — CHANGES_REQUESTED** [review]
Hi @monty16597, thanks for the PR!
Can you add the new TF tests that will pass with this change to the `terraform-tests.success.txt` please?

**Comment by monty16597** [comment]
Hey @bblommers ,
This PR is still in progress, I accidentally removed WIP from the head. I will update it once we are ready to merge.

**Comment by bblommers** [comment]
No worries @monty16597. I'll mark it as a draft then, to prevent accidental merges.

**Comment by monty16597** [comment]
Sounds Good, @bblommers  Thanks.

**Comment by monty16597** [comment]
Hello @bblommers ,
We are ready to merge the PR.

**Review by bblommers — APPROVED** [review]
Nice one - thanks @monty16597!

**Pull request merged**